### PR TITLE
Tidy up `test_memmap_implementation`

### PR DESCRIPTION
**Pull request by jakirkham**
* Test absence of key
* Assign data for easier checking
* Test presence of key before getting data

### CODE REVIEW

**Comment by jakirkham** [comment]
Planning on merging tomorrow if no comments

File: zict/tests/test_file.py [hunk 1]
```diff
@@ -43,11 +43,12 @@ def test_memmap_implementation(fn):
     z = File(fn, memmap=True)
     assert not z
 
-    z["x"] = b"123"
+    mv = memoryview(b"123")
```
Comment (jrbourbeau) [inline zict/tests/test_file.py:46]: Previously we were testing that we could add `bytes` to the mapping and get out a `memoryview`. Now we're doing `memoryview` in and `memoryview` out. Is the `bytes` case intended to be supported?

**Review by jrbourbeau — COMMENTED** [review]
Thanks @jakirkham! 

It's certainly outside the scope of this PR. But I noticed the `memmap` parameter isn't described in the `File` docstring -- if you have the bandwidth, would you mind adding a description for that option?

File: zict/tests/test_file.py (hunk 1 above)
Comment (jakirkham) [reply zict/tests/test_file.py:46]: We can try both if you like. Either should work as both `writelines` & `write` can handle any [Python Buffer Protocol object]( https://docs.python.org/3/c-api/buffer.html )

That particular code path is currently unaffected by the `memmap` flag

**Comment by jakirkham** [comment]
Sure added the documentation in PR ( https://github.com/dask/zict/pull/71 ).

It is worth noting we have an unused parameter, `mode`, which is currently documented ( https://github.com/dask/zict/issues/52 ).

**Review by jrbourbeau — APPROVED** [review]

**Pull request merged**